src/main/java/net/minecraftforge/common/extensions/IForgeItem.java

```diff
@@ -341,14 +341,15 @@ default void onArmorTick(ItemStack stack, Level level, Player player)
     default void onInventoryTick(ItemStack stack, Level level, Player player, int slotIndex, int selectedIndex)
     {
     	// For compatibility reasons we have to use non-local index values, I think this is a vanilla bug but lets maintain compatibility
-    	int vanillaIndex = slotIndex;
-    	if (slotIndex >= 36) {
-    		vanillaIndex -= 36;
-    		if (vanillaIndex >= 4)
-    			vanillaIndex -= 4;
-			else
-				onArmorTick(stack, level, player);
-    	}
+        var inv = player.getInventory();
+        int vanillaIndex = slotIndex;
+        if (slotIndex >= inv.items.size()) {
+            vanillaIndex -= inv.items.size();
+            if (vanillaIndex >= inv.armor.size())
+                vanillaIndex -= inv.armor.size();
+            else
+                onArmorTick(stack, level, player);
+        }
```

Comment:
Please preserve the formatting that was used prior.

Reply:
Ok